packages/paste-core/components/visual-picker/src/VisualPickerCheckbox.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {useUID} from '@twilio-paste/uid-library';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {BaseRadioCheckboxControl, BaseRadioCheckboxLabel} from '@twilio-paste/base-radio-checkbox';
+import {HiddenCheckbox, CheckboxIcon} from '@twilio-paste/checkbox';
+import {ScreenReaderOnly} from '@twilio-paste/screen-reader-only';
+import {useMergeRefs} from '@twilio-paste/utils';
+import {type BorderColor} from '@twilio-paste/style-props';
+
+import {VisualPickerCheckboxContext} from './VisualPickerContext';
+import {type VisualPickerCheckboxProps} from './types';
+
+export const VisualPickerCheckbox = React.forwardRef<HTMLInputElement, VisualPickerCheckboxProps>(
+  ({checked, element = 'VISUAL_PICKER_CHECKBOX', children, id, indeterminate, labelText, onChange, ...props}, ref) => {
+    const [isHovering, setIsHovering] = React.useState(false);
+
+    const visualPickerCheckboxGroupContext = React.useContext(VisualPickerCheckboxContext);
+    const helpTextId = useUID();
+    const labelId = useUID();
+    const checkboxId = id ? id : useUID();
+
+    const handleChange = React.useCallback(
+      (event: React.ChangeEvent<HTMLInputElement>): void => {
+        onChange(event);
+      },
+      [onChange]
+    );
+
+    // Prioritizing direct props values over whatever visualPickerCheckboxGroupContext passes down
+    const disabled = props.disabled != null ? props.disabled : visualPickerCheckboxGroupContext.groupIsDisabled;
+    const name = props.name != null ? props.name : visualPickerCheckboxGroupContext.name;
+    const hasError = props.hasError != null ? props.hasError : visualPickerCheckboxGroupContext.groupHasError;
+
+    const {groupHasError, groupIsDisabled} = visualPickerCheckboxGroupContext;
+
+    const internalRef = React.useRef<HTMLInputElement>();
+    const mergedRef = useMergeRefs(internalRef, ref) as React.Ref<HTMLInputElement>;
+
+    let pickerBorderColor = 'colorBorder' as BorderColor;
+    if (disabled || groupIsDisabled) pickerBorderColor = 'colorBorderWeaker';
+    else if (hasError || groupHasError) pickerBorderColor = 'colorBorderDestructive' as BorderColor;
+    else if (checked) pickerBorderColor = 'colorBorderPrimary' as BorderColor;
+
+    return (
+      <Box
+        element={element}
+        position="relative"
+        display="inline-flex"
+        alignItems={visualPickerCheckboxGroupContext.orientation === 'vertical' ? 'center' : 'flex-start'}
+        flexDirection="row"
+        backgroundColor={disabled ? 'colorBackgroundWeak' : undefined}
+        borderStyle="solid"
+        borderColor={pickerBorderColor}
+        borderWidth="borderWidth10"
+        borderRadius="borderRadius20"
```

Comment:
todo: We changed this to borderRadius30 today in crit to match card + file uploader